### PR TITLE
fix transcript db column type

### LINKED ISSUE #7: Transcription failed
logsays stt_helper | Failed to save transcript in DB: An exception occurred while executing a query: SQLSTATE[22001]: String data, right truncated: 7 ERROR: value too long for type character varying(255)

### PR DESCRIPTION
**Pull request by kyteinsky**
Fixes #7

### CODE REVIEW

**Review by bigcat88 — APPROVED** [review]
tested on MariaDB - works fine

**Comment by kyteinsky** [comment]
Thanks for the review @bigcat88 !  
Found out that changing the type of the column set it to `tinytext` when some text is present in the column before the migration and `longtext` if db was empty. This new fix works as expected and surprisingly does not get rid of the old transcripts from v1.0.0

**Review by marcelklehr — APPROVED** [review]

File: psalm.xml [hunk 1]
```diff
@@ -27,6 +27,7 @@
 		<UndefinedClass>
 			<errorLevel type="suppress">
 				<referencedClass name="OC\Hooks\Emitter" />
+				<referencedClass name="Doctrine\DBAL\Types\Type" />
```
Comment (nickvergessen) [inline psalm.xml:30]: you should not use this. please migrate to `OCP\DB\Types` instead :)

File: lib/Migration/Version1001Date20231229124844.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2023 Your name <your@email.com>
```
Comment (nickvergessen) [inline lib/Migration/Version1001Date20231229124844.php:6]: 🧐

File: lib/Migration/Version1001Date20231229124844.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * @author Your name <your@email.com>
```
Comment (nickvergessen) [inline lib/Migration/Version1001Date20231229124844.php:8]: 🧐

File: psalm.xml (hunk 1 above)
Comment (kyteinsky) [reply psalm.xml:30]: ah forgot to remove that from previous code that changed column's type.

File: lib/Migration/Version1001Date20231229124844.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	 * @param Closure(): ISchemaWrapper $schemaClosure
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		$schema = $schemaClosure();
+
+		$table = $schema->getTable('stt_helper_transcripts');
+		$table->dropColumn('transcript');
+		$table->addColumn('transcript', Types::TEXT, [
```
[the remark this answers is not in the document]
Comment (kyteinsky) [reply lib/Migration/Version1001Date20231229124844.php]: definitely works in a predictable manner, thanks!

File: lib/Migration/Version1001Date20231229124844.php (hunk 2 above)
Comment (kyteinsky) [reply lib/Migration/Version1001Date20231229124844.php]: oops :)

File: lib/Migration/Version1001Date20240109122003.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	 * @param IOutput $output
+	 * @param Closure(): ISchemaWrapper $schemaClosure
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		$schema = $schemaClosure();
+
+		$table = $schema->getTable('stt_helper_transcripts');
+		$table->addColumn('transcript', 'text', [
```
Comment (nickvergessen) [inline lib/Migration/Version1001Date20240109122003.php:54]:
```suggestion
		$table->addColumn('transcript', \OCP\DB\Types::TEXT, [
```
?

File: lib/Migration/Version1001Date20240109122003.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+		return $schema;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure(): ISchemaWrapper $schemaClosure
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
```
Comment (nickvergessen) [inline lib/Migration/Version1001Date20240109122003.php:66]: You can remove all those empty functions

File: lib/Migration/Version1001Date20240109122003.php (hunk 5 above)
Comment (kyteinsky) [reply lib/Migration/Version1001Date20240109122003.php:54]: sorry, that's just my IDE's helper tools acting funny

**Review by marcelklehr — APPROVED** [review]

**Pull request merged**